MapboxNavigation/Style.swift

```diff
@@ -209,6 +210,115 @@ public class ResumeButton: UIControl {
     }
 }
 
+/// :nodoc:
+@objc(MBSpeedLimitSign)
+open class SpeedLimitSign: UIView {
+    
+    private static let textInsets = UIEdgeInsets(top: 6, left: 6, bottom: 6, right: 6)
+    
+    var labels: [SpeedLimitLabel] = []
+    
+    let defaultText = NSLocalizedString("SPEED_LIMIT", bundle: .mapboxNavigation, value: "Speed Limit", comment: "Speed limit sign main text.")
+    var defaultTextSplit: [String] {
+        return defaultText.components(separatedBy: " ")
+    }
+    let paragraphStyle = NSMutableParagraphStyle()
+    let fontSize = UIFont.systemFontSize
+    let strings = NSMutableAttributedString()
+    let numberFormater = NumberFormatter()
+    let stackView = UIStackView()
+    
+    var speedLimit: SpeedLimit? {
+        didSet {
+            guard let speedLimit = speedLimit else { return }
+            guard let speedString = numberFormater.string(from: speedLimit.value as NSNumber) else { return }
+            
+            labels.removeAll()
+            
+            let defaultLabels = defaultTextSplit.map { (string: String) -> SpeedLimitLabel in
+                let label: SpeedLimitLabel = .forAutoLayout()
+                label.attributedText = NSMutableAttributedString(string: string.uppercased(), attributes: [
+                    .paragraphStyle: paragraphStyle,
+                    .font: UIFont.systemFont(ofSize: fontSize * 0.9)
+                    ])
+                return label
+            }
+            labels.append(contentsOf: defaultLabels)
+            
+            let speedLabel: SpeedLimitLabel = .forAutoLayout()
+            speedLabel.attributedText = NSMutableAttributedString(string: speedString, attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize * 2),
+                .paragraphStyle: paragraphStyle
+                ])
+            labels.append(speedLabel)
+            
+            let unitLabel: SpeedLimitLabel = .forAutoLayout()
+            unitLabel.attributedText = NSMutableAttributedString(string: "\n\(speedLimit.unit.localizedSpeedUnit)", attributes: [
+                .font: UIFont.boldSystemFont(ofSize: fontSize / 1.3),
+                .paragraphStyle: paragraphStyle
+                ])
+            labels.append(unitLabel)
+            
```

Comment:
@bsudekum I see a bit of code repetition in the 3 blocks of codes for `defaultLabels `, `speedLabel` and `unitLabel`. Could we keep it DRY?